File: 백준/Gold/1806. 부분합/부분합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {


        st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int s = Integer.parseInt(st.nextToken());
        int answer = n + 1;

        st = new StringTokenizer(br.readLine());
        int[] nums = new int[n];
        for (int i = 0; i < n; i++) {
            int num = Integer.parseInt(st.nextToken());
            nums[i] = num;
        }
        
        int left = 0, right = 0;
        int sum = nums[left];

        for (; left < n; left++) {
            right = Math.max(left, right);
            while (right < n-1 && sum < s) {
                right++;
                sum += nums[right];
            }
            if (sum >= s) {
                answer = Math.min(answer, right - left + 1);
            }
            sum -= nums[left];
        }

        System.out.println(answer > n ? 0 : answer);
    }
}

File: 백준/Gold/1806. 부분합/부분합.py
N, S = map(int, input().split())
nums = list(map(int, input().split()))
left, right = 0, 0
sum = 0
minLength = N + 1

while True:
    if sum < S:
        if right > N - 1:
            break
        sum += nums[right]
        right += 1
    else:
        minLength = min(right - left, minLength)
        if minLength == 1:
            break
        sum -= nums[left]
        left += 1

print(0 if minLength > N else minLength)